src/components/dar_dataset_table/DarDatasetTable.js

```diff
@@ -151,14 +152,32 @@ export const DarDatasetTable = (props) => {
   const [buckets, setBuckets] = useState([]);
 
   const {
-    collection, isLoading
+    collection, isLoading, summary
   } = props;
 
   const [isInitializing, setIsInitializing] = useState(true);
 
   const init = useCallback(async () => {
     try {
-      const { dars, datasets } = collection;
+      if (isEmpty(collection)) {
+        setBuckets([]);
+        return;
+      }
+      let { dars, datasets } = cloneDeep(collection);
+      datasets = datasets ? datasets.filter(dataset => summary.datasetIds.includes(dataset.dataSetId)) : null;
+      const darKeys = Object.keys(dars);
+      darKeys.forEach(darKey => {
+        if(intersection(summary.datasetIds, dars[darKey].datasetIds).length > 0) {
+          const electionKeys = Object.keys(dars[darKey].elections);
+          electionKeys.forEach(electionKey => {
+            if (!summary.datasetIds.includes(dars[darKey].elections[electionKey].datasetId)) {
+              delete dars[darKey].elections[electionKey];
+            }
```

Comment:
This looks like an unusual edge case ... writing this out so I'm clear on what's happening.

1. We're starting with a list of all datasets on the DAR
2. We then filter that on the dataset ids accessible to the chair/member as calculated by the back end
3. We then look at all elections on the DAR
4. We further restrict the visible datasets so that we are only seeing datasets for which we have elections.

We need step 4 because there could be datasets for which no elections have yet been created (not sure if this would also happen with canceled elections). I'd like to run this by Jonathan as it means we're purposefully NOT showing the user a dataset that the DAR is requesting only because no election has been created for it yet.

Reply:
Step 2 (line 167) filters the datasets to those included in the summary response.
Step 3 loops the individual DARs in the collection.  The existing code refers to these as DARs, so I kept the naming the same, but the history is apparent in this code.  :-) 
I then start to loop the individual DARs in the collection at 169.  The line at 173 is satisfied only when an election is found that **doesn't** pertain the the list of datasets we're filtering down to.  At that point, I'm removing that **election** information (#174) (not the dataset).

Reply:
Gotcha. Will rephrase my current understanding. There are two tasks here. 

1.  Filter the datasets to those that are visible to the chair/member.
2. Since we are displaying vote totals, we also have to filter elections that are not associated to the datasets in question.

We do have a method that should do the same thing, at least in theory, (`utils.filterCollectionElectionsByUser`) but I can see the utility here. I also think we could do a better job of packing up collection specific functions for common use.

Reply:
Great rephrasing, Greg!

Agreed on the collection specific function packaging.  I left it as-is for two reasons: 1) It's only current use is this function, 2) I wasn't sure if this should eventually make it into src/libs/utils.js **or** src/utils/DarCollectionUtils.js

Do you think we need to apply this logic elsewhere?

Reply:
> Great rephrasing, Greg!
> 
> Agreed on the collection specific function packaging.  I left it as-is for two reasons: 1) It's only current use is this function, 2) I wasn't sure if this should eventually make it into src/libs/utils.js **or** src/utils/DarCollectionUtils.js
> 
> Do you think we need to apply this logic elsewhere?

I think it's fine here, I don't see any other needs for this right now. Thank you!